CMakeLists.txt

```diff
@@ -484,7 +484,21 @@ if(${USE_TETGEN_NONFREE_TPL} MATCHES ON)
     endif(tetgen_FOUND)
 endif(${USE_TETGEN_NONFREE_TPL} MATCHES ON)
 
+# Checking if the MPI specific apps are compiled
```

Comment:
Metis does not require MPI any more (well my PR needs to be merged but hopefully thats only a matter of days)

In general I would prefer this mechanism to be in the apps themselves, same as we do with kratos_add_dependency

Other opinions?
@roigcarlo ?

Reply:
The problem is, MPI is check here, so if Trilinos is compiled and MPI is not selected it will fail. If the check is done later it would be too late

Reply:
I would put the check in the app itself, as other requirements.

I would not create a cmake module for that as its simply an if, and trilinos is the only app that has that dependency. But if you want to do it I won't block it.

Reply:
But here not only the check is done, is activating it in case it is not

Reply:
that's the problem, we should not assume that someone wants to enable MPI because it has enabled trilinos. It could be what the user wanted or it could simply be because the user did not remember to remove it.

In general I prefer this kind of errors to show up, because then is easier to diagnose. 

I am thinking on (for example) the case of someone trying to compile HDF5Application with the serial lib and having the Trilinos app enabled as well. In that case it would crash because the libhdf5-mpi not found, even when MPI is off because it has been auto-activated by trilinos, which is not obvious unless you know what is happening underneath.

Reply:
Okay, in that case I will put an error in trilinos instead of activating, because Trilinos app requires MPI 100%

Reply:
Changes suggested applied